src/TerminalFighter/Settings/Settings.cpp

```diff
@@ -1,44 +1,232 @@
+    SettingsSection section,
+    std::vector<std::string> keys,
+    double& value) const {
+
+    YAML::Node node = load_node(section, keys);
+
+    if (!node.IsScalar()) {
+        LOG(ERROR) << "SettingsGroup " << vec_to_str(keys)
+                   << " from section " << static_cast<char>(section)
+                   << " is not double";
+        return false;
+    }
+
+    try {
+        value = node.as<double>();
+    } catch (YAML::InvalidNode e) {
+        LOG(ERROR) << "Could not load setting " << vec_to_str(keys)
+                   << " from section " << static_cast<char>(section)
+                   << ", " << e.msg;
+        return false;
+    } catch (YAML::RepresentationException e) {
+        LOG(ERROR) << "Couldn't load " << vec_to_str(keys) << " as double. " << e.msg;
+        return false;
+    }
+
+    return true;
 }
 
-const YAML::Node& Settings::asset_paths() const {
-    return asset_paths_;
+bool Settings::load_bool(
+    SettingsSection section,
+    std::vector<std::string> keys,
+    bool& value) const {
+
+    YAML::Node node = load_node(section, keys);
+
+    if (!node.IsScalar()) {
+        LOG(ERROR) << "SettingsGroup " << vec_to_str(keys)
+                   << " from section " << static_cast<char>(section)
+                   << " is not boolean";
+        return false;
+    }
+
+    try {
+        value = node.as<bool>();
+    } catch (YAML::InvalidNode e) {
+        LOG(ERROR) << "Could not load setting " << vec_to_str(keys)
+                   << " from section " << static_cast<char>(section)
+                   << ", " << e.msg;
+        return false;
+    } catch (YAML::RepresentationException e) {
+        LOG(ERROR) << "Couldn't load " << vec_to_str(keys) << " as bool. " << e.msg;
+        return false;
+    }
+
+    return true;
+}
+
+bool Settings::load_str_map(
+    SettingsSection section,
+    std::vector<std::string> keys,
+    std::map<std::string, std::string>& value) const {
+
+    YAML::Node node = load_node(section, keys);
+
+    if (!node.IsMap()) {
+        LOG(ERROR) << "SettingsGroup " << vec_to_str(keys)
+                   << " from section " << static_cast<char>(section)
+                   << " is not a map";
+        return false;
+    }
+
+    value.clear();
+
+    for (YAML::const_iterator it = node.begin(); it != node.end(); ++it) {
+        try {
+            value[it->first.as<std::string>()] = it->second.as<std::string>();
+        } catch (YAML::InvalidNode e) {
+            LOG(ERROR) << "Could not load setting " << vec_to_str(keys)
+                       << " from section " << static_cast<char>(section)
+                       << ", " << e.msg;
+            return false;
+        } catch (YAML::RepresentationException e) {
+            LOG(ERROR) << "Couldn't load " << vec_to_str(keys) << " as map<str,str>. " << e.msg;
+            return false;
+        }
+    }
+
+    return true;
 }
+
+/* PRIVATE FUNCTIONS */
+
+int Settings::reload_setting(SettingsSection section) {
```

Comment:
I like this. Gives the person making the GUI configuration options on how to reload settings